core/src/bitmap/bitmap_data.rs

```diff
@@ -332,18 +332,21 @@ mod wrapper {
 
         // Provides access to the underlying `BitmapData`. If a GPU -> CPU sync
         // is in progress, waits for it to complete
-        pub fn sync(&self) -> GcCell<'gc, BitmapData<'gc>> {
+        pub fn sync(&self, renderer: &mut dyn RenderBackend) -> GcCell<'gc, BitmapData<'gc>> {
             // SAFETY: The only fields that can store gc pointers are `avm2_object` and `dirty_callbacks`,
             // which we don't update here. Ideally, we would refactor this so that
             // `BitmapData` doesn't contain any gc pointers, allowing us to use a normal
             // `RefCell` instead of a `GcCell`.
             let mut write = unsafe { self.0.borrow_mut() };
             match std::mem::replace(&mut write.dirty_state, DirtyState::Clean) {
                 DirtyState::GpuModified(sync_handle, bounds) => {
-                    sync_handle
-                        .retrieve_offscreen_texture(Box::new(|buffer, buffer_width| {
-                            copy_pixels_to_bitmapdata(&mut write, buffer, buffer_width, bounds)
-                        }))
+                    renderer
+                        .resolve_sync_handle(
+                            sync_handle,
+                            Box::new(|buffer, buffer_width| {
+                                copy_pixels_to_bitmapdata(&mut write, buffer, buffer_width, bounds)
+                            }),
+                        )
                         .expect("Failed to sync BitmapData");
```

Comment:
There's no way this panic can be hit on the webgl/canvas backends, right?

Reply:
Correct